src/components/Conversion/ConversionNOB.vue

```diff
@@ -0,0 +1,145 @@
+            class="mt-5"
+            autocomplete="chrome-off"
+            label="Enter Nature of Business"
+            v-model="naicsText"
+            counter="300"
+            :rules="naicsRules"
+            @change="onChangeNaics($event)"
+          >
+          </v-text-field>
+          <div id="nob-confirm-container" class="mb-2">
+            <v-btn large color="primary" id="nob-done-btn" class="mr-2"
+              @click="onSubmitClicked()"
+            >
+              <span>Done</span>
+            </v-btn>
+            <v-btn large outlined color="primary" id="nob-cancel-btn"
+              @click="onCancelClicked()"
+            >
+              <span>Cancel</span>
+            </v-btn>
+          </div>
+        </div>
+
+        <div v-if="!onEditMode" class="summary-block d-flex justify-space-between align-center">
+          <span id="naics-summary">{{ naicsSummary }}</span>
+          <v-btn text color="primary" id="nob-change-btn" @click="onChangeClicked()">
+            <v-icon small>mdi-pencil</v-icon>
+            <span>Change</span>
+          </v-btn>
+        </div>
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Action, Getter } from 'vuex-class'
+import { Component, Vue } from 'vue-property-decorator'
+import { ActionBindingIF } from '@/interfaces/'
+import { NaicsIF } from '@bcrs-shared-components/interfaces/'
+import { isEqual } from 'lodash'
+
+@Component({})
+export default class NatureOfBusiness extends Vue {
+  @Getter getCurrentNaics!: NaicsIF
+  @Getter hasNatureOfBusinessChanged!: boolean
+
+  @Action setNaics!: ActionBindingIF
+
+  // local variables
+  private onEditMode = false
+  private naicsRules = []
+  private naicsText = ''
+
+  /** Show naics value, description or (Not Entered) upon first render */
+  get naicsSummary (): string {
+    const code = this.getCurrentNaics.naicsCode
+    const desc = this.getCurrentNaics.naicsDescription
+    let summary = '(Not Entered)'
+    if (code && desc) {
+      this.naicsText = this.hasNatureOfBusinessChanged ? this.naicsText : `${code} - ${desc}`
+      summary = `${code} - ${desc}`
+    } else if (desc) {
+      this.naicsText = desc
+      summary = desc
+    }
+    return summary
+  }
+
+  /** The naics data on record for the business. */
+  get originalNaics (): NaicsIF {
+    return {
+      naicsCode: this.getCurrentNaics.naicsCode,
+      naicsDescription: this.getCurrentNaics.naicsDescription
+    }
+  }
+
+  /** Monitor naics text change */
+  private onChangeNaics (naicsText:string): void {
+    this.naicsRules = naicsText.length > 300 ? ['Maximum 300 characters reached'] : []
+  }
+
+  /** Called when user has clicked the Change button. */
+  onChangeClicked (): void {
+    this.onEditMode = true
+  }
+
+  /** Submited when user has clicked the Done button. */
+  onSubmitClicked (): void {
+    if (this.naicsText.length > 300) {
+      this.naicsRules = ['Maximum 300 characters reached']
```

Comment:
In some cases, the input component(s) would be wrapped in a v-form, and you could simply validate the form here and verify that it's valid. See https://vuetifyjs.com/en/api/v-form/#functions.

In this case, you can actually do the same thing with the text field component. You'll need to declare a ref to it, and then you can call `myComponentRef.validate()` and check that the result is True.

Basically, use the functionality already built in the component to declare validation rules and check if the component is valid.

Reply:
Looks good!